h2o-algos/src/main/java/hex/ensemble/StackedEnsembleModel.java

```diff
@@ -344,6 +476,32 @@ void checkAndInheritModelProperties() {
         throw new H2OIllegalArgumentException("Base model is not supervised: "+aModel._key.toString());
       }
 
+      if (require_consistent_training_frames) {
+        if (trainingFrameRows < 0) trainingFrameRows = _parms.train().numRows();
+        Frame aTrainingFrame = aModel._parms.train(); // can be null when aModel was imported
+        if (aTrainingFrame == null) { // The aModel was probably loaded from a different session.
+          Log.warn("Training frame of model \"" + aModel._key + "\"  is null. " +
+                  "Assuming it's same as the training frame of this Stacked Ensemble.");
+          aTrainingFrame = DKV.getGet(aModel._output._cross_validation_holdout_predictions_frame_id);
+          if (aTrainingFrame == null) {
+            throw new H2OIllegalArgumentException("Base model \"" + aModel._key + "\" is missing cross validation holdout predictions.");
+          }
+        }
+        if (trainingFrameRows != aTrainingFrame.numRows())
+          throw new H2OIllegalArgumentException("Base models are inconsistent: they use different size (number of rows) training frames."
+                  +" Found: "+trainingFrameRows+" (StackedEnsemble) and "+aTrainingFrame.numRows()+" (model "+k+").");
+      }
+
+      if (cv_required_on_base_model) {
+        // If we don't have a fold_column require:
+        // nfolds > 1
+        // nfolds consistent across base models
+        if (aModel._parms._fold_column == null && aModel._parms._nfolds < 2)
+          throw new H2OIllegalArgumentException("Base model does not use cross-validation: " + aModel._parms._nfolds);
+        if (!aModel._parms._keep_cross_validation_predictions)
+          throw new H2OIllegalArgumentException("Base model does not keep cross-validation predictions: " + aModel._parms._nfolds);
+      }
+
```

Comment:
getting lost here. Is it related with this PR?

Reply:
if this is clean-up / improvement, can you please move to different PR?
For a small cleanup, I wouldn't mind, but looks like there was also some logic added.

Reply:
Sorry, not completely related to this PR. It's just moved couple lines up from the `if (retrievedFirstModelParams) {` so we check all the base models not just starting with the second one.

Reply:
to be clear, if you think it's justified to include the changes in this PR, please go on (probably better this than a FIXME).
From a reviewer perspective, I just noticed that some lines were moved, which is totally fine, but I also noticed that some logic was changed without explanation: as it didn't look immediately related with the PR (but maybe it is), I just expect the author to add a small comment in the diff view to inform reviewers, and avoid them scratching their head wondering "why this change?".

Reply:
If I would think it's justified I would try to argue a little bit more but this PR is quite complex already. The part with FIXME was a result of experimentation with saving and loading AutoML models that I did when you were on vacation.

I would create a separate jira for that functionality (and tests), if that's ok with you...

Reply:
perfect!